## Naive_bayes kfolds CV

### Telecom data :

### Independent variable is Gender_M, TT_GT_100_Y and Depende variable is Response_Y

In [8]:
import pandas as pd
telecom = pd.read_csv("E:/class/R _Cource/dataset class/Telecom.csv")

dum_telecom = pd.get_dummies(telecom, drop_first=True)
dum_telecom.head()

Gender_M  TT_gt_100_Y  Response_Y
0         0            1           0
1         1            0           0
2         1            0           0
3         0            1           1
4         0            0           0

In [9]:

#from sklearn.model_selection import StratifiedKFold
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score


In [10]:

#from sklearn.naive_bayes import BernoulliNB
from sklearn.naive_bayes import MultinomialNB

X = dum_telecom.iloc[:,0:2]
y = dum_telecom.iloc[:,2]



In [11]:

kfold = KFold(n_splits=5, random_state=42)
#kfold = StratifiedKFold(n_splits=5, random_state=42)
multinomial = MultinomialNB()
results = cross_val_score(multinomial, X, y, cv=kfold, scoring='roc_auc')
print(results)
print("ROC AUC: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))


[0.796875   0.8447205  0.9479638  0.92410714 0.835     ]
ROC AUC: 86.97% (5.70%)


### Roc AUC value is 86.97% which is very very good for k is equal to 5